dev/release/post-11-bump-versions-test.rb

```diff
@@ -276,6 +276,34 @@ def test_version_post_tag
       next if hunks.empty?
       expected_changes << {hunks: hunks, path: path}
     end
+    if release_type == :major
+      expected_changes << {
+        hunks: [
+          [
+            "-[![Go Reference](https://pkg.go.dev/badge/github.com/apache/arrow/go/v#{@snapshot_major_version}.svg)](https://pkg.go.dev/github.com/apache/arrow/go/v#{@snapshot_major_version})",
+            "+[![Go Reference](https://pkg.go.dev/badge/github.com/apache/arrow/go/v#{@next_major_version}.svg)](https://pkg.go.dev/github.com/apache/arrow/go/v#{@next_major_version})",
+          ],
```

Comment:
Does `README.md` need to be added to [the glob above](https://github.com/apache/arrow/pull/37780/files#diff-ed4591fc42e3d74c25735a1cf85f0dff430649978397ea4cf94cd156ec6c5530R238)?

I thought something like this might be required:
```diff
     end
 
-    Dir.glob("go/**/{go.mod,*.go,*.go.*}") do |path|
+    Dir.glob("go/**/{go.mod,*.go,*.go.*,README.md}") do |path|
       if path == "go/arrow/doc.go"
         expected_changes << {
           path: path,
```

Reply:
If we want to use this approach, we also need to update the `Dir.glob` content. See https://github.com/apache/arrow/pull/37780/commits/d9f7f63155627b00b501bf2f024ba486911ba32f for details.